crates/fdev/src/commands.rs

```diff
@@ -143,32 +139,49 @@ pub async fn update(config: UpdateConfig, other: BaseConfig) -> Result<(), anyho
         StateDelta::from(buf).into()
     };
     let request = ContractRequest::Update { key, data }.into();
-    execute_command(request, other).await
+    execute_command(request, other, config.address, config.port).await
 }
 
 async fn execute_command(
     request: ClientRequest<'static>,
     other: BaseConfig,
+    address: IpAddr,
+    port: u16,
 ) -> Result<(), anyhow::Error> {
     let mode = other.mode;
-    let paths = other.paths.build()?;
-    let contract_store = ContractStore::new(paths.contracts_dir(mode), DEFAULT_MAX_CONTRACT_SIZE)?;
-    let delegate_store = DelegateStore::new(paths.delegates_dir(mode), DEFAULT_MAX_DELEGATE_SIZE)?;
-    let secret_store = SecretsStore::new(paths.secrets_dir(mode))?;
-    let state_store = StateStore::new(Storage::new(&paths.db_dir(mode)).await?, MAX_MEM_CACHE)?;
-    let rt =
-        freenet::dev_tool::Runtime::build(contract_store, delegate_store, secret_store, false)?;
-    let mut executor = Executor::new(state_store, || Ok(()), OperationMode::Local, rt, None)
-        .await
-        .map_err(|err| anyhow::anyhow!(err))?;
 
-    executor
-        .handle_request(ClientId::FIRST, request, None)
-        .await
-        .map_err(|err| {
-            tracing::error!("{err}");
-            err
-        })?;
+    let target = match mode {
+        OperationMode::Local => {
+            if !address.is_loopback() {
+                return Err(anyhow::anyhow!(
+                    "invalid ip: {address}, expecting a loopback ip address in local mode"
+                ));
+            }
+            SocketAddr::new(address, port)
+        }
+        OperationMode::Network => {
+            if address.is_loopback() {
```

Comment:
This ain't correct, we always go through the local node, and it will take care of targeting where it should target remotely. But apps always go via the local node and don't attempt sending puts to remote ws APIs (that would never work since we don't expose the ws api over the network)